jwst/assign_wcs/nircam.py

```diff
@@ -198,6 +198,13 @@ def tsgrism(input_model, reference_files):
     # begin at 0,0, so no need to translate the crpix to full frame
     # because they already are in full frame coordinates.
     xc, yc = (input_model.meta.wcsinfo.siaf_xref_sci, input_model.meta.wcsinfo.siaf_yref_sci)
+
```

Comment:
@nden Should 1 be subtracted from `xc` and from `yc`?

Reply:
I don't remember if these values, which come from the SIAF, are 1-indexed or 0-indexed. @nden do you know?

Reply:
In `extract_tso_object` in extract_2d/grisms.py, 1 is subtracted from these values to get `source_xpos` and `source_ypos`, so it looks to me as if one or the other of these functions should be changed.

Reply:
If the transforms are defined using 0-based coordinates, yes. And normally we would want them 0-based. But I will have to double check that with the INS team. I suggest we merge this as is and file a ticket (I can file it). I also see use of `CRVAL1/2`  below which probably isn't necessary, we should use `RA/DEC_REF`.
I will review the tsgrism code after I finish with wfss.

Reply:
Sorry I didn't see the previous replies. If extract_2d is subtracting 1 and `source_x/ypos` are used then everything is correct. I'd prefer to subtract it here though when this is defined.
@hbushouse The SIAF values are 1-based.

Reply:
So, based on @nden comments, we should modify line 200 to subtract 1 from siaf_xref_sci and siaf_yref_sci when assigning their values to xc, yc, and then ... (see further comments below)

Reply:
@nden I'm not sure what you mean by "when this is defined."  Do you mean the transform that's defined in function `tsgrism` in assign_wcs/nircam.py?  It looks to me as if the wcs created by `extract_tso_object` in extract_2d/grisms.py is just the one from assign_wcs/nircam but with shifts and spectral order selection added.  That is, the wcs in extract_2d adds to rather than replaces the one from assign_wcs.  Am I misinterpreting the code in extract_2d?

Reply:
I think this goes beyond the scope of this PR. I am refactoring the code and will look into it in detail when I get to this function. My comment above was not meant to be a requirement for this PR, I just noticed and was surprised by the use of CRVALs.

Reply:
That's fine with me, to confine the scope of this PR to only adding the raising of errors when the meta values aren't available, and not try to fix any other issues involving the actual use of xref_sci or crval1/2.